src/main/java/com/hillel/items_exchange/service/impl/UserServiceImpl.java

```diff
@@ -93,4 +96,15 @@ public void registerNewUser(UserRegistrationDto userRegistrationDto, BCryptPassw
             log.warn("IN UserServiceImpl (registerNewUser): user: {} not registered", user);
         }
     }
+
+    @Override
+    public Optional<UserDto> getByUsernameOrEmail(String usernameOrEmail) {
+        return userRepository.findByUsername(usernameOrEmail).map(this::mapUserToDto);
+    }
```

Comment:
I'm sorry that didn't mentioned earlier, but I think this method should returns `Optional<User>`. I can't use the return type UserDto, it doesn't fit to my logic.

Reply:
I heard you. I'll talk to Roma about it.

Reply:
@lubchik-14 you can do with Optional whatever you want. :smiley:

Reply:
@rpkyrych 🤔 but UserDto does not have for example children or phones. And I have to work with User instance and its relations. How do I do this? Can you give me an example, please.

Reply:
Discussed internally with @lubchik-14. The topic is resolved.